### PR TITLE
Add Permissions Docs Compiler

### PR DESCRIPTION
**Pull request by igor-egorov**
### Description of the Change

The change introduces a way to compile docs permissions page automatically.
Now, `permissions.rst` should NOT be modified manually.

The compiler uses `csv` file as a source of permissions' description, notable information, and examples description.
The compiler automatically makes references to the glossary for the terms occurred in permissions' descriptions (like wiki engine). The same term within a single description would not be transformed to a reference more than once.
The compiler makes a reference to commands that are related to corresponding permissions.
The compiler automatically includes listings of python examples.

### Benefits

It is easier to maintain the long list of permissions in any table/sheet editor than edit very long `rst` source file.

Permissions page format can be easily changed via applying changes to compiler script.

### Possible Drawbacks 

Potentially no beautiful architecture inside compiler scripts. It may become a subject of refactoring later on.

### Usage Examples or Tests

Please check already compiled page here https://html-drqyakiigy.now.sh/maintenance/permissions
(Navigation menu may be broken, most likely due to internal url rewrite rules of now.sh)

The other way is to compile docs locally - please check `docs/README.md` and issue `make permissions` before `make html`.

### CODE REVIEW

**Comment by l4l** [comment]
Would be nice to give a reference at `core_concepts/glossary.html#permission` and maybe some other places. Also I'm not sure, that maintenance is a good place

**Comment by igor-egorov** [comment]
Thanks! I will add a link into the glossary. 

We consider "Maintenance" section good enough because tuning of permissions is a kind of maintenance process. Nevertheless, feel free to share your vision of a better place for the page.

File: docs/permissions_compiler/rst.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+                break
+
+    return (begin, end)
+
+
+def excerpt(permission):
+    """
+
+    :param permission:
+    :param example_description: list of strings
```
Comment (lebdron) [inline docs/permissions_compiler/rst.py]: There is no such parameter, maybe remove this line?

**Comment by igor-egorov** [comment]
There is no reason to wait CI. Merged without waiting.

**Pull request merged**